### ISSUE TITLE
Using a tileLoadFunction with a MVT VectorTile source causes only some tiles to render

### ISSUE THREAD

**Issue by lasselaakkonen** [openlayers/openlayers#8276]
When using layers like this:

```
layers: [
    new ol.layer.Tile({
      source: new ol.source.OSM()
    }),
    new ol.layer.VectorTile({
      source: new ol.source.VectorTile({
        format: new ol.format.MVT(),
        url: 'dummy-for-tileLoadFunction',
        tileLoadFunction: function(tile) { ... },
      }),
    }),
  ]
```

Some tiles are rendered, but tiles stop rendering after zooming/panning. Both OSM and VectorTile tiles stop rendering.

If an actual `url` is used, without a `tileLoadFuction`, then tiles render as they should.

**Reproducing the issue**
- https://codepen.io/anon/pen/KeywPo
- Zoom in to some city
- Zoom out
- Zoom in to some other city
-> Tiles will stop rendering at some point. This happens every time, sometimes already when zooming in to the first city.

Reproduced with OL 4.6.5 on macOS Chrome + Firefox.

**Comment by lasselaakkonen** [comment]
Thank you for the fast reply.

Always calling `setFeatures()` and `setProjection()` fixes the issue in my real use case as well.

I had the issue with an actual implementation and it seemed that I could simplify the issue to the codepen I created. I also got sidetracked, because some tiles were rendered, then suddenly rendering stopped.

I ran into that scenario, because my tile source (right now testing with https://github.com/urbica/tessella) does not return anything, if there are is no vector data in the tile. And in my code I only called `setFeatures()` and `setProjection()` when there was data.

So now `tileLoadFunction()` is called for all tiles on the map, but I can not call e.g. `tile.setProjection(format.readProjection(data));`, because there is no data, but I know the projection, so I can still call it with the known projection.
